relay-server/src/metrics_extraction/spans/mod.rs

```diff
@@ -303,6 +310,26 @@ fn sanitized_span_description(
     Some(sanitized)
 }
 
+/// Trims the given span description to ensure
+/// [`relay-metrics::AggregatorService`] doesn't drop it for being too long.
+///
+/// If the description is short, it remains unchanged. If it's long, this method
+/// truncates it to the maximum allowed length and sets the last character to
+/// `*`. The maximum length is determined by [`MaxChars::TagValue`].
+///
+/// Note [`relay-metrics::AggregatorService`] is configured independently and
+/// doesn't use this type; this method uses it for simplicity, and because at
+/// the time of writing this the values are the same. If the limit this type
+/// provides is longer, this method may be ineffective.
+fn truncate_span_description(description: String) -> String {
+    let mut truncated = description;
+    if bytecount::num_chars(truncated.as_bytes()) > MaxChars::TagValue.limit() {
```

Comment:
The metrics aggregator seems to limit the number of bytes in the string, so why are we checking the number of _characters_ here instead? https://github.com/getsentry/relay/blob/83a435e0754df148184c4ad642fb8a33eba3a469/relay-metrics/src/aggregation.rs#L1629